spec/integration/workers/indexer/amend_worker_spec.rb

```diff
@@ -8,34 +9,42 @@
   let(:updates) { { "title" => "New title" } }
   let(:cluster_count) { Clusters.count }
 
+  before do
+    Sidekiq::Worker.clear_all
+  end
+
   it "amends documents" do
-    commit_document(index_name, document)
+    Sidekiq::Testing.fake! do
+      commit_document(index_name, document)
 
-    request = stub_request_to_publishing_api(content_id)
+      request = stub_request_to_publishing_api(content_id)
 
-    described_class.new.perform(index_name, link, updates)
+      described_class.new.perform(index_name, link, updates)
 
-    doc_with_updates = document.merge(updates)
+      doc_with_updates = document.merge(updates)
 
-    assert_requested request, times: cluster_count
-    expect_document_is_in_rummager(doc_with_updates, id: link, index: index_name)
+      assert_requested request, times: cluster_count
+      expect_document_is_in_rummager(doc_with_updates, id: link, index: index_name)
```

Comment:
If the fake testing only enqueues jobs and doesn't execute them, why is the updated document found in searc here?

Reply:
because the class is called directly here. I suppose in this case the fake! might not be necessary because the index isn't locked and the infinite loop that the fake! is meant to avoid won't happen here - I'll remove this one.